packages/cli/src/services/generator/target/parsing.ts

```diff
@@ -7,12 +7,26 @@ export const parseName = (name: string): Promise<HasName> => Promise.resolve({ n
 
 export const parseEvent = (event: string): Promise<HasEvent> => Promise.resolve({ event })
 
-export const parseFields = (fields: Array<string>): Promise<HasFields> =>
-  Promise.all(fields.map(parseField)).then((fields) => ({ fields }))
+export const parseFields = (fields: Array<string>): Promise<HasFields> => {
+  return Promise.all(fields.map(parseField)).then((fields) => {
+    let usedFieldNames: string[] = []
+    let duplicated: string | undefined = undefined
+    fields.forEach((field) => {
+      if (usedFieldNames.includes(field.name) && duplicated === undefined) {
+        duplicated = field.name
+      }
+      usedFieldNames.push(field.name)
+    })
+    if (duplicated !== undefined) {
+      return Promise.reject(fieldDuplicatedError(duplicated))
+    }
+    return { fields }
+  })
+}
```

Comment:
Just a suggestion: if you work with promises, consider using an `async` function. This way, the code inside the function can throw normal errors or return values without the need of nesting everything inside promises, making it more readable. 

Also, as you are finding duplicates, you can leverage the native `Set` object that automatically removes duplicates for you and allows you to do set operations, like intersections. This is a suggestion for the code that combines the two things I have mentioned here:

```typescript
const parseFields = async (fields: Array<string>): Promise<HasFields> => { // <-- Notice the "async" prefix
  const parsedFields = fields.map(parseField) 
  const parsedFieldNames = parsedFields.map((field) => field.name) // Get only the field names
  const uniqueFieldNames = new Set(parsedFieldNames)               // Create a set of unique field names
  const duplicates = parsedFieldNames.filter((field) => !uniqueFieldNames.delete(field)) // This "intersection" extracts all duplicates
  if (duplicates.length > 0) {
      // This "throw" is converted automatically to a Promise.reject()
      // (BTW, we should change this to the correct error message)
      throw new Error("Duplicates: " + duplicates.join(', ')) 
  }
  return parsedFields // This "return" is automatically converted to a Promise.resolve()
}
```

Reply:
Good suggestions by the way. I need to study more in depth the await, async, Promise syntax. I have applied the changes @alvaroloes